src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -755,8 +756,14 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
                             .getString(AuthenticationConstants.Browser.RESPONSE_ERROR_MESSAGE);
                     Logger.v(TAG, "Error info:" + errCode + " " + errMessage + " for requestId: "
                             + requestId + correlationInfo);
+                    
+                    ADALError adalError = ADALError.SERVER_INVALID_REQUEST;
+                    if (errCode.equalsIgnoreCase("Code:" + String.valueOf(WebViewClient.ERROR_FAILED_SSL_HANDSHAKE))) {
```

Comment:
put ADAL error code into bundle and just use it,
no need to parse and compare strings, when enums are serializable